PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -710,6 +712,35 @@ function Start-CopyJob(
     }
 }
 
+# 
+# Utility wrapper for invoking commands by opening sessions  
+# for each of the cluster nodes and preserving the session  
+# to be deleted after use. 
+# 
+
+function Invoke-SddcCommonCommand ( 
+    [ValidateNotNullOrEmpty()][string[]] $ClusterNodes = @(), 
+    [string] $JobName,
+	[scriptblock] $InitBlock, 
+    [scriptblock] $ScriptBlock 
+    ) 
+{ 
+	$Job        = @() 
+	$Sessions   = @() 
+	$SessionIds = @() 
+
+	$Sessions = New-PSSession -ComputerName $ClusterNodes 
+	Invoke-Command -Session $Sessions $InitBlock
+	$Job      = Invoke-Command -Session $Sessions -AsJob -JobName $JobName -ScriptBlock $ScriptBlock 
+
+	foreach ($s in $Sessions) { 
+		$SessionIds += $s.Id 
+	} 
+
+	$Job | Add-Member -NotePropertyName ActiveSessions -NotePropertyValue $SessionIds  

```

Comment:
You could add -PassThru and the updated Job object will come out onto the pipeline for return. No need to return it with an explicit statement, unless you prefer the idiom.